File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/entities/permissions/entityPermission.ts
/// <reference path="../base/iBaseEntity.ts" />
/// <reference path="../base/baseEntity.ts" />
/// <reference path="../../entities.ts" />
/// <reference path="../../../helpers.ts" />
/// <reference path="./permissionAccessTypeEnum.ts" />

namespace mdBusinessLogic.dataAccess.entities.permissions {
    export class entityPermission {
        public Object: entitiesEnum;
        public Entity: entitiesEnum;
        public AccessTypes: Array<permissionAccessTypeEnum>;

        constructor() {
            this.AccessTypes = new Array<permissionAccessTypeEnum>();
        }
    }
}

File: MD.CMS.BusinessLogic.Core/DataAccess/Entities/Permissions/EntityPermission.cs
using System.Collections.Generic;
namespace MD.CMS.BusinessLogic.Core.DataAccess.Entities.Permissions
{
    public class EntityPermission
    {
        #region Attributes
        private Tools.BaseDataAccess.Plugins.Core.Mapping.Entities _object;
        private Tools.BaseDataAccess.Plugins.Core.Mapping.Entities _entity;
        private HashSet<PermissionAccessTypeEnum> _accessTypes;
        #endregion

        #region Properties
        public Tools.BaseDataAccess.Plugins.Core.Mapping.Entities Entity { get => _entity; set => _entity = value; }
        public HashSet<PermissionAccessTypeEnum> AccessTypes 
        { 
            get
            {
                if(_accessTypes == null)
                {
                    _accessTypes = new HashSet<PermissionAccessTypeEnum>();
                }
                return _accessTypes;
            }
            set => _accessTypes = value; 
        }
        public Tools.BaseDataAccess.Plugins.Core.Mapping.Entities Object { get => _object; set => _object = value; }
        #endregion
    }
}
